packages/flutter_tools/test/general.shard/base/bot_detector_test.dart

```diff
@@ -132,6 +136,21 @@ void main() {
       expect(await azureDetector.isRunningOnAzure, isFalse);
     });
 
+    testWithoutContext('isRunningOnAzure returns false when the http request times out', () {
+      FakeAsync().run((FakeAsync time) {
+        when(mockHttpClient.getUrl(any)).thenAnswer((_) {
+          final Completer<HttpClientRequest> completer = Completer<HttpClientRequest>();
+          return completer.future;  // Never completed to test timeout behavior.
+        });
+
+        unawaited(azureDetector.isRunningOnAzure.then((bool onBot) {
```

Comment:
The structure of this test is a little odd, because it seems like this expectation isn't guaranteed to be hit before the test body finishes. I would rearrange it a bit, maybe something like this?

```
FakeAsync().run((FakeAsync time) async {
  ...
  final Future<bool> isRunningOnAzure = azureDetector.isRunningOnAzure;
  time.elapse(const Duration(seconds: 2));

  expect(await isRunningOnAzure, true);
});
```

Reply:
Done.